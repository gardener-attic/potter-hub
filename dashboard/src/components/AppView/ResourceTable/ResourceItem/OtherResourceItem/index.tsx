import OtherResourceItem from "./OtherResourceItem";

export default OtherResourceItem;
