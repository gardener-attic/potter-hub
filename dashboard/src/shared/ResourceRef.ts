import { Kube } from "./Kube";
import { ResourceKind } from "./ResourceKinds";
import { IResource } from "./types";

// ResourceRef defines a reference to a namespaced Kubernetes API Object and
// provides helpers to retrieve the resource URL
class ResourceRef {
  public apiVersion: string;
  public kind: ResourceKind;
  public name: string;
  public namespace: string;
  public isOnResourceCluster: boolean;

  // Creates a new ResourceRef instance from an existing IResource. Provide
  // defaultNamespace to set if the IResource doesn't specify a namespace.
  // TODO: add support for cluster-scoped resources, or add a ClusterResourceRef
  // class.
  constructor(r: IResource, defaultNamespace?: string, isOnResourceCluster?: boolean) {
    this.apiVersion = r.apiVersion;
    this.kind = r.kind;
    this.name = r.metadata.name;
    const namespace = r.metadata.namespace || defaultNamespace;
    if (!namespace) {
      throw new Error(`Namespace missing for resource ${this.name}, define a default namespace`);
    }
    this.namespace = namespace;
    isOnResourceCluster
      ? (this.isOnResourceCluster = isOnResourceCluster)
      : (this.isOnResourceCluster = false);
    return this;
  }

  // Gets a full resource URL for the referenced resource
  public getResourceURL() {
    return Kube.getResourceURL(
      this.apiVersion,
      Kube.resourcePlural(this.kind),
      this.isOnResourceCluster,
      this.namespace,
      this.name,
    );
  }

  public watchResourceURL() {
    return Kube.watchResourceURL(
      this.apiVersion,
      Kube.resourcePlural(this.kind),
      this.isOnResourceCluster,
      this.namespace,
      this.name,
    );
  }

  public getResource() {
    return Kube.getResource(
      this.apiVersion,
      Kube.resourcePlural(this.kind),
      this.isOnResourceCluster,
      this.namespace,
      this.name,
    );
  }

  // Opens and returns a WebSocket for the requested resource. Note: it is
  // important that this socket be properly closed when no longer needed. The
  // returned WebSocket can be attached to an event listener to read data from
  // the socket.
  public watchResource() {
    return Kube.watchResource(
      this.apiVersion,
      Kube.resourcePlural(this.kind),
      this.isOnResourceCluster,
      this.namespace,
      this.name,
    );
  }
}

export default ResourceRef;
