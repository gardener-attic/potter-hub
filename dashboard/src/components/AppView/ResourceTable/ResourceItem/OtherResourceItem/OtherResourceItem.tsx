import * as React from "react";
import ResourceRef from "shared/ResourceRef";

export interface IOtherResourceProps {
  resource: ResourceRef;
}

export const OtherResourceColumns: React.SFC = () => {
  return (
    <React.Fragment>
      <th className="col-6">NAME</th>
      <th className="col-6">KIND</th>
    </React.Fragment>
  );
};

const OtherResourceItem: React.SFC<IOtherResourceProps> = props => {
  const { resource } = props;
  return (
    <tr key={`otherResources/${resource.kind}/${resource.name}`} className="flex">
      <td className="col-6">{resource.name}</td>
      <td className="col-6">{resource.kind}</td>
    </tr>
  );
};

export default OtherResourceItem;
