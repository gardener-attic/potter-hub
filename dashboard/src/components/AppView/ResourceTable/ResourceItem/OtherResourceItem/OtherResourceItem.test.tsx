import { shallow } from "enzyme";
import * as React from "react";

import ResourceRef from "shared/ResourceRef";
import OtherResourceItem from "./OtherResourceItem";

it("renders a ConfigMap", () => {
  const cm = {
    name: "foo",
    kind: "ConfigMap",
  } as ResourceRef;
  const wrapper = shallow(<OtherResourceItem resource={cm} />);
  expect(wrapper).toMatchSnapshot();
});
