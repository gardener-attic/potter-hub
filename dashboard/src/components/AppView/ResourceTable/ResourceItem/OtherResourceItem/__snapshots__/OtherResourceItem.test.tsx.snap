// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ConfigMap 1`] = `
<tr
  className="flex"
  key="otherResources/ConfigMap/foo"
>
  <td
    className="col-6"
  >
    foo
  </td>
  <td
    className="col-6"
  >
    ConfigMap
  </td>
</tr>
`;
